ui/packages/catalog-ui-search/src/main/webapp/component/dropdown/hover-preview/dropdown.hover-preview.view.js

```diff
@@ -13,55 +13,62 @@
  *
  **/
 /*global define*/
-const Marionette = require('marionette')
-const _ = require('underscore')
-const $ = require('jquery')
+import React from 'react'
+
 const DropdownView = require('../dropdown.view')
-const template = require('./dropdown.hover-preview.hbs')
 const ComponentView = require('../../hover-preview/hover-preview.view.js')
-const properties = require('../../../js/properties.js')
-const _merge = require('lodash/merge')
 const Common = require('../../../js/Common.js')
 const user = require('../../singletons/user-instance.js')
 
 module.exports = DropdownView.extend({
-  template: template,
-  className: 'is-hover-preview',
-  componentToShow: ComponentView,
-  events: {
-    'mouseenter button': 'handleMouseEnter',
-    'mouseleave button': 'handleMouseLeave',
-    'click button': 'handleClick',
-  },
-  handleMouseEnter: function() {
-    if (user.getHoverPreview()) {
-      this.model.open()
+  events: {}, // remove base events
+  template() {
+    const metadataThumbnail = this.modelForComponent
+      .get('metacard')
+      .get('properties')
+      .get('thumbnail')
+    const model = this.model
+
+    const openThumbnailInNewWindow = () =>
+      window.open(Common.getImageSrc(metadataThumbnail))
+    const onMouseEnter = () => user.getHoverPreview() && model.open()
+    const onMouseLeave = () => model.close()
+    const onImageError = () => {
+      this.imageLoadError = true
+      this.render()
     }
-  },
-  handleMouseLeave: function() {
-    this.model.close()
-  },
-  handleClick: function() {
-    window.open(
-      Common.getImageSrc(
-        this.options.modelForComponent
-          .get('metacard')
-          .get('properties')
-          .get('thumbnail')
-      )
+
+    return (
+      (metadataThumbnail && (
```

Comment:
❓ Could we please make this a conditional block? Relying on short-circuiting makes it pretty hard to figure out what's actually going on here.

Reply:
We can certainly revisit this in the future, though I would prefer to stick to functional parts of this PR as it's coming up on a month.